### PR TITLE
Add PowerBI embedded module

### PR DESCRIPTION
**Pull request by kevindelmont**
## PR Checklist

---

- [x] I have added example(s) inside the [./examples/] folder
- [x] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Add [powerbi embedded](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/r/powerbi_embedded.html.markdown) module

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Testing

Use the example I provided

### CODE REVIEW

**Comment by arnaudlh** [comment]
Thanks for your contribution @kevindelmont, your examples only creates a RG, I think you are missing a couple of variables declarations. Could you please update?  thanks

File: examples/powerbi_embedded/100-powerbi_embeded/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+resource_groups = {
+  rg1 = {
+    name   = "powerbi"
+    region = "region1"
+  }
+}
+
+powerbi_embedded = {
```
Comment (kevindelmont) [inline examples/powerbi_embedded/100-powerbi_embeded/configuration.tfvars:15]: Hi @arnaudlh, the example of creating the powerbi resource is already there.

File: modules/powerbi_embedded/module.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+resource "azurecaf_name" "powerbi" {
```
Comment (arnaudlh) [inline modules/powerbi_embedded/module.tf:1]:
```suggestion
data "azurecaf_name" "powerbi" {
```

File: modules/powerbi_embedded/module.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  prefixes      = var.global_settings.prefixes
+  random_length = var.global_settings.random_length
+  clean_input   = true
+  passthrough   = var.global_settings.passthrough
+  use_slug      = var.global_settings.use_slug
+}
+
+
+resource "azurerm_powerbi_embedded" "powerbi" {
+  name                = azurecaf_name.powerbi.result
```
Comment (arnaudlh) [inline modules/powerbi_embedded/module.tf:13]:
```suggestion
  name                = data.azurecaf_name.powerbi.result
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
@kevindelmont for the example to work, you will need to add ```powerbi_embedded = var.powerbi_embedded``` in /examples/module.tf and declare the variable also in /examples.

Please also use the data source instead of resource for name, to improve plan-time data, as suggested.

File: examples/powerbi_embedded/100-powerbi_embeded/configuration.tfvars (hunk 1 above)
Comment (arnaudlh) [reply examples/powerbi_embedded/100-powerbi_embeded/configuration.tfvars:15]: yes it is, but its not working :)

File: modules/powerbi_embedded/module.tf (hunk 3 above)
Comment (kevindelmont) [reply modules/powerbi_embedded/module.tf]: I updated

**Review by arnaudlh — APPROVED** [review]
LGTM

**Comment by arnaudlh** [comment]
There seem to be a bug in the naming provider, I have filled https://github.com/aztfmod/terraform-provider-azurecaf/issues/248 to track it. In the meantime, the feature is still useable with a ```passthrough = true``` in the ```global_settings``` section.

**Pull request merged**